mission_control/navigator_missions/nav_missions/detect_deliver.py

```diff
@@ -79,58 +82,84 @@ def circle_search(self):
         #  yield self.navigator.move.stop()
         #  fprint("Ended Circle Search", title="DETECT DELIVER",  msg_color='green')
 
+    def update_shape(self, shape_res, normal_res, tf):
+       self.identified_shapes[(shape_res.Shape, shape_res.Color)] = self.get_shape_pos(normal_res, tf)
+
+    def correct_shape(self, shape):
+        return (self.Color == "ANY" or self.Color == shape.Color) and (self.Shape == "ANY" or self.Shape == shape.Shape)
+
     @txros.util.cancellableInlineCallbacks
-    def align_to_target(self):
-        self.markers = MarkerArray()
-        while True:
-            if (yield self.is_found()):
-                if (yield self.get_normal()):
-                    break
+    def search_shape(self):
+        shapes = yield self.get_shape()
+        if shapes.found:
```

Comment:
Navigator checks for this by default, you probably don't want that exception to be thrown right,